pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroup.java

```diff
@@ -341,7 +341,11 @@ protected BytesAndMessagesCount updateLocalQuota(ResourceGroupMonitoringClass mo
         oldBMCount = monEntity.quotaForNextPeriod;
         try {
             monEntity.quotaForNextPeriod = newQuota;
-            this.resourceGroupPublishLimiter.update(newQuota);
+            if (monClass == ResourceGroupMonitoringClass.Publish) {
+                this.resourceGroupPublishLimiter.update(newQuota);
+            } else {
```

Comment:
Since we are early-quitting for anything except Publish (see line 329), I didn't see why this check for non-Publish is required again in line 346.

Reply:
when I made this change in my local sandbox, I did not have the changes you committed in commit ID '60a569823293'. Now that the check is added in  commit ID '60a569823293', this change is no longer required. I will update the code.